### DISCUSSION TITLE
Need help to configure nvimtree

### DISCUSSION THREAD

**Discussion by alan-rizzo** [NvChad/NvChad#659] [Q&A] [answered]
I downloaded NvChad yesterday and I want to change nvimtree setup.

And I do not have idea how to do it.

```lua
M.mappings.plugins = {
   nvimtree = {
      filters = {
         dotfiles = true,
         custom = {"__pycache__"}
      },
   },
}
```
I wrote that on **lua/custom/chadrc.lua**

**Comment by siduck** [comment, marked as answer]
https://nvchad.com/docs/config/plugins

**Reply by aniwid** [reply]
Also wondering how to change the nvimtree config, link above is returning a 404. I edited the `.local/share/nvim/lazy/NvChad/lua/nvchad/configs/nvimtree.lua` file but NvChad won't update now until I remove my changes. So wondering where to put my custom config changes. If someone could answer in more than 3 words like most of the docs that would be awesome too, cheers.

**Reply by siduck** [reply]
@b33fstrogan0ff updated the link

**Reply by aniwid** [reply]
Thanks, but that doesn't really tell me how to do anything. I'm assuming the plugins folder it's referring to is `$HOME/.config/nvim/lua/plugins`? 

So how do I set the view width of nvimtree to 50 using that? Create an `nvimtree.lua` file in that plugins folder, or edit the `init.lua` in there already and add something like the following:
```
{
    "nvimtree/nvimtree",
    opts = {
        view = {
            width = 50
        }
    }
}
```
?

Also how do I install new plugins? There never seems to be any file paths mentioned anywhere so never know WHERE this stuff is supposed to happen, including the Lazy docs, just a bunch of code snippets, doesn't say where.

**Reply by siduck** [reply]
@b33fstrogan0ff you can add it anywhere within plugin/ even plugins/init.lua's table, your code looks right

**Reply by aniwid** [reply]
Didn't work unfortunately :(

**Reply by aniwid** [reply]
Nevermind, figured it out! 

Needed a dash between `nvim` and `tree`, so I created an `nvim-tree.lua` file inside my `$HOME/.config/nvim/lua/plugins` directory, then added the following:
```
return {
    "nvim-tree/nvim-tree.lua",
    opts = {
        view = {
            width = 50
        }
    }
}
```
That works!
